src/library/compiler/ir_interpreter.cpp

```diff
@@ -470,8 +470,22 @@ class interpreter {
         }
     }
 
+    void check_system() {
+        try {
+            lean::check_system("interpreter");
+        } catch (stack_space_exception & ex) {
+            sstream ss;
+            ss << ex.what() << "\n";
+            ss << "interpreter stacktrace:\n";
+            for (unsigned i = 0; i < m_call_stack.size(); i++) {
+                ss << "#" << (i + 1) << " " << m_call_stack[m_call_stack.size() - i - 1].m_fn << "\n";
+            }
+            throw throwable(ss);
```

Comment:
@leodemoura Note that I'm changing the exception type, but there is no other `catch (stack_space_exception)` in the current code.